.jenkins/infrastructure/build_azure_managed_images.Jenkinsfile

```diff
@@ -94,78 +94,119 @@ def buildLinuxManagedImage(String os_type, String version, String managed_image_
+                            "JENKINS_VNET_NAME=${jenkins_vnet_name}",
+                            "VM_RG_NAME=${vm_rg_name}",
+                            "REGION=${REGION}",
+                            "VM_NAME=${vm_name}",
+                            "AZURE_IMAGE_ID=${azure_image_id}"]) {
+                        sh '''
+                            SUBNET_ID=\$(az network vnet subnet show \
+                                --resource-group ${JENKINS_RG_NAME} \
+                                --name ${JENKINS_SUBNET_NAME} \
+                                --vnet-name ${JENKINS_VNET_NAME} --query id -o tsv)
+                            az vm create \
+                                --resource-group ${VM_RG_NAME} \
+                                --location ${REGION} \
+                                --name ${VM_NAME} \
+                                --size Standard_DC4s_v2 \
+                                --os-disk-size-gb 128 \
+                                --subnet \$SUBNET_ID \
+                                --admin-username ${SSH_USERNAME} \
+                                --admin-password ${SSH_PASSWORD} \
+                                --image ${AZURE_IMAGE_ID} \
+                                --public-ip-address ""
+                        '''
+                    }
                 }
+            }
 
-                stage("Deploy VM") {
-                    def deploy_script = """
-                        ${az_login_script}
-
-                        VM_DETAILS=\$(az vm show --resource-group ${vm_rg_name} \
-                                                --name ${vm_name} \
-                                                --show-details)
-
-                        az vm run-command invoke \
-                            --resource-group ${vm_rg_name} \
-                            --name ${vm_name} \
-                            --command-id EnableRemotePS
-
-                        PRIVATE_IP=\$(echo \$VM_DETAILS | jq -r '.privateIps')
-                        rm -f ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "[windows-agents]" >> ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "\$PRIVATE_IP" >> ${WORKSPACE}/scripts/ansible/inventory/hosts-${launch_configuration}
-                        echo "ansible_user: ${JENKINS_USER_NAME}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "ansible_password: ${JENKINS_USER_PASSWORD}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "ansible_winrm_transport: ntlm" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
-                        echo "launch_configuration: ${launch_configuration}" >> ${WORKSPACE}/scripts/ansible/inventory/host_vars/\$PRIVATE_IP
+            stage("Deploy VM") {
+                checkout([$class: 'GitSCM',
```

Comment:
In what cases do you checkout GitSCM?

Reply:
I'm not sure I understand this question. We checkout the openenclave repo when the files are necessary. In this case we need the Ansible files to that we can run Ansible to set up the VM.